ksp_plugin/interface_iterator.cpp

```diff
@@ -0,0 +1,52 @@
+#include "ksp_plugin/interface.hpp"
+
+#include <vector>
+
+#include "journal/method.hpp"
+#include "journal/profiles.hpp"
+#include "ksp_plugin/plugin.hpp"
+
+namespace principia {
+
+using ksp_plugin::LineSegment;
+
+namespace interface {
+
+bool principia__IteratorAtEnd(Iterator const* const iterator) {
+  journal::Method<journal::IteratorAtEnd> m({iterator});
+  return m.Return(CHECK_NOTNULL(iterator)->AtEnd());
+}
+
+void principia__IteratorDelete(Iterator** const iterator) {
+  journal::Method<journal::IteratorDelete> m({iterator}, {iterator});
+  TakeOwnership(iterator);
+  return m.Return();
+}
+
+void principia__IteratorIncrement(Iterator* const iterator) {
+  journal::Method<journal::IteratorIncrement> m({iterator});
+  CHECK_NOTNULL(iterator)->Increment();
+  return m.Return();
+}
+
+XYZSegment principia__IteratorGetXYZSegment(Iterator const* const iterator) {
+  journal::Method<journal::IteratorGetXYZSegment> m({iterator});
+  CHECK_NOTNULL(iterator);
+  auto const* typed_iterator =
```

Comment:
`not_null<TypedIterator<std::vector<LineSegment<World>> const*> const`, to detect `dynamic_cast` failure.